relay-general/src/protocol/event.rs

```diff
@@ -538,13 +538,6 @@ pub struct Event {
 }
 
 impl Event {
-    pub fn get_transaction_source(&self) -> &TransactionSource {
-        self.transaction_info
-            .value()
-            .and_then(|info| info.source.value())
-            .unwrap_or(&TransactionSource::Unknown)
-    }
```

Comment:
This function does not distinguish between `None` and `Some(Unknown)`, even though it makes a semantic difference -> better delete it.